src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/ConstantExpectedAnalyzer.cs

```diff
@@ -0,0 +1,566 @@
+                    context.ReportDiagnostic(valueDiagnostic);
+                }
+            }
+        }
+
+        protected abstract void RegisterAttributeSyntax(CompilationStartAnalysisContext context);
+
+        protected void OnParameterWithConstantExpectedAttribute(IParameterSymbol parameter, Action<Diagnostic> reportAction)
+        {
+            if (!ValidateConstantExpectedParameter(parameter, out ImmutableArray<Diagnostic> diagnostics))
+            {
+                foreach (var diagnostic in diagnostics)
+                {
+                    reportAction(diagnostic);
+                }
+            }
+        }
+
+        private static bool TryCreateConstantExpectedParameter(IParameterSymbol parameterSymbol, [NotNullWhen(true)] out ConstantExpectedParameter? parameter)
+        {
+            if (!TryGetConstantExpectedAttributeData(parameterSymbol, out var attributeData))
+            {
+                parameter = null;
+                return false;
+            }
+
+            switch (parameterSymbol.Type.SpecialType)
+            {
+                case SpecialType.System_Char:
+                    return UnmanagedHelper<char>.TryCreate(parameterSymbol, attributeData, char.MinValue, char.MaxValue, out parameter);
+                case SpecialType.System_Byte:
+                    return UnmanagedHelper<ulong>.TryCreate(parameterSymbol, attributeData, byte.MinValue, byte.MaxValue, out parameter);
+                case SpecialType.System_UInt16:
+                    return UnmanagedHelper<ulong>.TryCreate(parameterSymbol, attributeData, ushort.MinValue, ushort.MaxValue, out parameter);
+                case SpecialType.System_UInt32:
+                    return UnmanagedHelper<ulong>.TryCreate(parameterSymbol, attributeData, uint.MinValue, uint.MaxValue, out parameter);
+                case SpecialType.System_UInt64:
+                    return UnmanagedHelper<ulong>.TryCreate(parameterSymbol, attributeData, ulong.MinValue, ulong.MaxValue, out parameter);
+                case SpecialType.System_UIntPtr:
+                    return UnmanagedHelper<ulong>.TryCreate(parameterSymbol, attributeData, uint.MinValue, uint.MaxValue, out parameter);
+                case SpecialType.System_SByte:
+                    return UnmanagedHelper<long>.TryCreate(parameterSymbol, attributeData, sbyte.MinValue, sbyte.MaxValue, out parameter);
+                case SpecialType.System_Int16:
+                    return UnmanagedHelper<long>.TryCreate(parameterSymbol, attributeData, short.MinValue, short.MaxValue, out parameter);
+                case SpecialType.System_Int32:
+                    return UnmanagedHelper<long>.TryCreate(parameterSymbol, attributeData, int.MinValue, int.MaxValue, out parameter);
+                case SpecialType.System_Int64:
+                    return UnmanagedHelper<long>.TryCreate(parameterSymbol, attributeData, long.MinValue, long.MaxValue, out parameter);
+                case SpecialType.System_IntPtr:
```

Comment:
Seems IntPtr and UIntPtr cannot have a const values, [so according to @tannergooding's comment](https://github.com/dotnet/roslyn-analyzers/pull/5766/files#r919280091) it should not be covered

Reply:
nint/nuint support removed